toolbox/swiftsim/particle_splitting/generate_splitting_information.py

```diff
@@ -270,9 +280,23 @@ def get_descendant_particle_ids(old_snapshot_data, new_snapshot_data):
 
             # If we have a new tree, all new particle IDs have as their progenitor the
             # particle ID that originated this unique tree.
-            progenitor_id = tree_progenitor_ID
+            # NOTE: Disabled because SWIFT runs had incorrect ParticleProgenitorIDs
+            # progenitor_id_old = tree_progenitor_ID
 
             new_ids = new_snapshot_data['particle_ids'][tree_index]
+
+            # We get the ID that has all 0s in its split tree (it retained the ID of
+            # the original particle)
+            progenitor_id = new_ids[new_snapshot_data["trees"][tree_index] == 0]
+
+            # Print out a warning if no progenitor ID was found... We cannot do much more
+            # than that.
+            if len(progenitor_id) == 0:
+                local_no_progenitors_found += 1
+                continue
```

Comment:
Could we an assert here that `len(progenitor_id) < 2`?

Reply:
Not sure I understand. What do you mean? Progenitor ID should exactly be 1, and it cannot be larger than that. If it is zero, we have a problem.

Reply:
I agree it shouldn't ever be greater than 1, so the assert would just be to make sure that's definitely the case.

Reply:
Done.